api/python/t4/data_transfer.py

```diff
@@ -604,16 +605,30 @@ def get_size_and_meta(src):
         raise NotImplementedError
     return size, meta, version
 
-def calculate_sha256(src_list, total_size):
+def calculate_sha256(src_list, sizes):
+    total_size = sum(sizes)
     lock = Lock()
 
     with tqdm(desc="Hashing", total=total_size, unit='B', unit_scale=True) as progress:
-        def _process_url(src):
+        def _process_url(src, size):
             src_url = urlparse(src)
             hash_obj = hashlib.sha256()
             if src_url.scheme == 'file':
                 path = pathlib.Path(parse_file_url(src_url))
+
                 with open(path, 'rb') as fd:
+                    current_file_size = pathlib.Path(path).stat().st_size
+                    if current_file_size != size:
+                        old_size = humanize.naturalsize(size)
+                        new_size = humanize.naturalsize(current_file_size)
+                        warnings.warn(
+                            f"Expected the package entry at {src!r} to be {old_size} in size, but "
+                            f"found an object which is {new_size} instead. This indicates that "
+                            f"the content of the file changed in between when you included this "
+                            f" entry in the package (via set or set_dir) and now. This should be "
```

Comment:
I'd just say "this is not supported"

Reply:
It's not quite right to say `this is not supported` though, because we're throwing a warning, not an error; so clearly we *do* support this. Unless we want this to be an error?